src/tools/pub/package-layout.md

```diff
@@ -313,10 +313,17 @@ tests) but you should never import from another package's `lib/src` directory.
 Those files are not part of the package's public API, and they might change in
 ways that could break your code.
 
-When you use libraries from within your own package, even code in `src`, you
-can (and should) still use `package:` to import them. For example:
+When you use libraries from within your own package, you should use:
+ * `package:` when [reaching inside or outside `lib/`](https://dart.dev/guides/language/effective-dart/usage#dont-allow-an-import-path-to-reach-into-or-out-of-lib) (lint: [_avoid_relative_lib_imports_](https://dart.dev/tools/linter-rules#avoid_relative_lib_imports))
+ * [Prefer relative imports](https://dart.dev/guides/language/effective-dart/usage#prefer-relative-import-paths) otherwise.
```

Comment:
Switches to use relative links.

```suggestion
When you use libraries from within your own package, you should use:

 * `package:` when [reaching inside or outside `lib/`][] (lint: [_avoid_relative_lib_imports_][]).
 * [Prefer relative imports][] otherwise.
 
 [reaching inside or outside `lib/`]: /guides/language/effective-dart/usage#dont-allow-an-import-path-to-reach-into-or-out-of-lib
 [_avoid_relative_lib_imports_]: /tools/linter-rules#avoid_relative_lib_imports
 [Prefer relative imports]: /guides/language/effective-dart/usage#prefer-relative-import-paths
```

Reply:
One tweak: The [intro to the list should be a sentence](https://developers.google.com/style/lists#introductory-sentences-for-lists), and the [list items should be parallel](https://developers.google.com/style/lists#parallel-syntax). Something like this:

How you import libraries from within your own package depends on the locations of the libraries:

* When [reaching...] (lint: ...), use `package:`.
* Otherwise, [prefer relative imports].